packages/ember-data/lib/system/model/states.js

```diff
@@ -417,6 +417,10 @@ createdState.uncommitted.rollback = function(record) {
   record.transitionTo('deleted.saved');
 };
 
+createdState.uncommitted.pushedData = function(record) {
+  record.transitionTo('loaded.saved');
+};
+
```

Comment:
Looks like inside `buildRecord` the newly created record is in the `root.loading` state so this code doesn't get hit.